tests/testsuite/host.rs

```diff
@@ -0,0 +1,107 @@
+use ansi_term::{Color, Style};
+use std::io;
+
+use crate::common;
+use crate::common::TestCommand;
+
+// TODO: test where hostname contains invalid UTF, triggering invalid hostname pattern
+
+#[test]
```

Comment:
Just to make sure I'm understanding correctly: it looks like these could all be run as unit tests from within the module, if it weren't for the fact that you want to test things with the actual hostname/configs? (Excellent suite of tests, by the way)

Is there any way at all that these tests could fail if run in a production environment? Non-ignored tests failing will stop certain installation methods, so I'd like to be extra-sure. Looking over them once, my gut feeling is "no", but I might have missed something.

Reply:
Yes they could be run there, I just assumed the testsuite directory was the correct place to put these.
I do not believe they could fail unless `gethostname` panics or returns invalid UTF. I'm changing the tests to not fail if `gethostname` returns invalid UTF.

Reply:
No, this is the correct place for them. I was just curious as to whether I was missing anything.